lib/srv/discovery/fetchers/db/db.go

```diff
@@ -128,16 +125,15 @@ func (f *AWSFetcherFactory) MakeFetchers(ctx context.Context, matchers []types.A
 			for _, makeFetcher := range makeFetchers {
 				for _, region := range matcher.Regions {
 					fetcher, err := makeFetcher(awsFetcherConfig{
-						AWSClients:                      f.cfg.CloudClients,
-						Type:                            matcherType,
-						AssumeRole:                      assumeRole,
-						Labels:                          matcher.Tags,
-						Region:                          region,
-						Integration:                     matcher.Integration,
-						DiscoveryConfigName:             discoveryConfigName,
-						AWSConfigProvider:               f.cfg.AWSConfigProvider,
-						IntegrationCredentialProviderFn: f.cfg.IntegrationCredentialProviderFn,
-						redshiftClientProviderFn:        f.cfg.RedshiftClientProviderFn,
+						AWSClients:               f.cfg.CloudClients,
+						Type:                     matcherType,
+						AssumeRole:               assumeRole,
+						Labels:                   matcher.Tags,
+						Region:                   region,
+						Integration:              matcher.Integration,
+						DiscoveryConfigName:      discoveryConfigName,
+						AWSConfigProvider:        f.cfg.AWSConfigProvider,
```

Comment:
`AWSConfigProvider` is an interface. I wonder if we can just create one that does all the integration stuff, something like 
```
AWSConfigProvider: newAWSConfigProviderWithIntegration(matcher.Integration),
```
Then we can move integration stuff out of `awsconfig`, maybe a separate util packet or to awsoidc.

Do we see any usage that integration name will change per `GetConfig` call?

Reply:
on a second thought though, above suggest will skip the cache